### DISCUSSION TITLE
Memory not release in DuckDB 0.6.0

### DISCUSSION THREAD

**Discussion by ntapsrigowri** [duckdb/duckdb#5418] [Q&A]
Thanks for some cool updates in latest DuckDB version. Memory management has improved significantly. I am using DuckDB to store about 100mil records, I create CSV files and load data in to duckDB (using C code) and later generate parquet files based on duckDB data and run various query over the parquet files(using node JS). I'm running this operation inside an Ubuntu container and DuckDB 0.6.0;I have noticed that even after all of this process is complete, memory usage of my container is stuck at 1GB(P.S: without DuckDB operataions, the memory usage is back to ~100MB) 
Any thoughts on what could be going wrong?

Some refs: 
<img width="440" alt="Screenshot 2022-11-19 at 11 35 12 PM" src="https://user-images.githubusercontent.com/47638406/202865199-1082896b-fcde-42f4-aef2-e5af892fa261.png">

Linux version:"Ubuntu 20.04.5 LTS (Focal Fossa)"
Node version: v16.18.1

top
top - 17:56:56 up 17 days, 14:02,  0 users,  load average: 0.19, 0.37, 0.72
Tasks:   4 total,   1 running,   3 sleeping,   0 stopped,   0 zombie
%Cpu(s):  1.3 us,  1.8 sy,  0.0 ni, 96.9 id,  0.0 wa,  0.0 hi,  0.1 si,  0.0 st
MiB Mem :  15884.9 total,   2780.5 free,   7729.7 used,   5374.7 buff/cache
MiB Swap:   2048.0 total,   2043.1 free,      4.9 used.   7798.7 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
    1 root      20   0    3968   1548   1304 S   0.0   0.0   0:00.04 entrypoint.sh
    _### **8 root      20   0   12.6g   1.0g  31984 S   0.0   6.6  25:11.81 node**_
 9292 root      20   0    4232   2248   1668 S   0.0   0.0   0:00.04 bash
 9302 root      20   0    6112   1832   1300 R   0.0   0.0   0:00.17 top

**Comment by Mytherin** [comment]
DuckDB keeps tables cached in memory until the user specified memory limit. You can close the database if you want to release that memory, or alternatively lower the memory limit.

**Reply by ntapsrigowri** [reply]
What is the default memory limit? I do close the db connection after running my query(attached below). Is there a way to force clear the cache programatically using nodejs?

async function runDuckDbQuery(query, dbName) {
    const db = new duckdb.Database(dbName);
    const response = new Promise(resolve => {
        db.all(query, (err, queryResponse) => {
            resolve(queryResponse);
        });
    });
    db.close();
    return response;
}

**Reply by Mytherin** [reply]
The default memory limit is 80% of system memory - you can query this using `PRAGMA database_size`, and change it using `SET memory_limit`. If you are closing the database system and memory still remains then that might be an issue with the specific (usage of the) client, but it is hard to say without having access to any code or a reproducible example.

**Reply by ntapsrigowri** [reply]
i'm not sure i completely understood how configuring memory_limit would help? Data would still be cached and amount of memory allocated in mem_limit is held by duckdb isnt it? 
Issue is probably with nodejs client; I have attached a sample code snippet and data csv file
If no duckdb queries are run and print a simple log statement; memory consumption is 141MB, if I run duckDB queries, memory consumption is 149MB

 
[DuckDBTest.zip](https://github.com/duckdb/duckdb/files/10049001/DuckDBTest.zip)

**Reply by Mytherin** [reply]
@Mause could you perhaps have a look to see what is going on here w.r.t. memory consumption?

**Reply by ntapsrigowri** [reply]
Hello @Mause and @Mytherin , Do you have some update for me on this thread? 
Thanks

**Reply by Mause** [reply]
> Hello @Mause and @Mytherin , Do you have some update for me on this thread? 
> Thanks

I'm honestly having trouble reliably reproducing it - any chance you could share a script or Dockerfile to reproduce it?

**Reply by ntapsrigowri** [reply]
[DuckDBTest.zip](https://github.com/duckdb/duckdb/files/10088802/DuckDBTest.zip)
Hey! Yes, csv data file and nodejs script attached
